src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```
@@ -0,0 +1,181 @@
+
+                        String caseNumber = getCaseNumber(dailyListHearing.getDefendants(), dailyListEntity, dailyListHearing);
+
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName, String.valueOf(sitting.getCourtRoomNumber()),
+                                caseNumber, dailyListHearing.getHearingDetails().getHearingDate()
+                        );
+
+                        CourtCaseEntity courtCase = hearing.getCourtCase();
+                        addJudges(sitting, hearing);
+                        addDefendants(courtCase, dailyListHearing.getDefendants());
+                        addProsecutions(courtCase, dailyListHearing.getProsecution().getAdvocate().getPersonalDetails().getName());
+                        addDefenders(courtCase, dailyListHearing.getDefendants());
+
+                        courtCase.addHearing(hearing);
+
+                        hearingRepository.saveAndFlush(hearing);
+                        caseRepository.saveAndFlush(courtCase);
+                    }
+                }
+            } else {
+                statusType = JobStatusType.PARTIALLY_PROCESSED;
+                log.error("Unregistered courthouse " + courtHouseName + " daily list entry has not been processed");
+            }
+        }
+        dailyListEntity.setStatus(statusType.name());
+    }
+
+    private String getCaseNumber(List<Defendant> defendants, DailyListEntity dailyListEntity, Hearing hearing) {
+        // CPP don't provide case id, use URN
+        if (String.valueOf(SourceType.CPP).equalsIgnoreCase(dailyListEntity.getSource())) {
+            if (defendants.isEmpty()) {
+                return hearing.getCaseNumber();
+            } else {
+                String urn = defendants.get(0).getUrn();
+                if (StringUtils.isEmpty(urn)) {
+                    log.error("Hearing not added - HearingInfo does not contain a URN value");
```

should the status be PARTIALLY_PROCESSED in this scenario?
Maybe this should throw an error that is caught by the calling method?

---

i think we should be marking as PARTIALLY_PROCESSED, regardless of what legacy is doing, @davet1985  what do you think?

---

I am happy to make the change I was just worried if legacy was doing something extra with these hearings.